pkg/secrets/secrets.go

```diff
@@ -37,12 +39,13 @@ func GetPrivateKey(secret kubeTypes.NamespacedName, c client.Client) ([]byte, er
 }
 
 // GenerateUserData generates the desired value of userdata secret.
-func GenerateUserData(publicKey ssh.PublicKey) (*core.Secret, error) {
+func GenerateUserData(platformType config.PlatformType, publicKey ssh.PublicKey) (*core.Secret, error) {
 	pubKeyBytes := ssh.MarshalAuthorizedKey(publicKey)
 	if pubKeyBytes == nil {
 		return nil, errors.Errorf("failed to retrieve public key using signer")
 	}
-
+	metadataKey := GetMetadataKeyForPlatform(platformType)
```

Comment:
i think we should proabably leave the metadata key the same for all platforms unless we have a reason to have different settings. the MAPI controllers are looking for the value `userData`, so i think we can keep it the same as previously. that will reduce the number of changes we need to make across components.

Reply:
Makes sense.